### PR TITLE
[RW-13586][risk=no] Add maintenance text for compliance training

### PR DESCRIPTION
**Pull request by evrii**
image

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx [hunk 1]
```diff
@@ -25,25 +25,32 @@ import { serverConfigStore } from 'app/utils/stores';
 import defaultServerConfig from 'testing/default-server-config';
 import { ProfileApiStub } from 'testing/stubs/profile-api-stub';
 
-const findHelpText = () => {
-  return screen.findByText(
+const queryHelpText = () => {
```
Comment (evrii) [inline ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx:28]: Unrelated cleanup

File: ui/src/app/pages/access/compliance-training-module-card-title.tsx [hunk 2]
```diff
@@ -43,7 +45,17 @@ export const ComplianceTrainingModuleCardTitle = ({
         Complete <AoU /> {trainingTitle}
       </div>
       {showHelpText && (
-        <p>Navigate to "My Courses" and select "{courseTitle}"</p>
+        <>
+          {blockComplianceTraining ? (
+            <p>
+              Our training system is conducting scheduled maintenance from
+              October 15th through October 18th. Please return after our
```
Comment (evrii) [inline ui/src/app/pages/access/compliance-training-module-card-title.tsx:52]: Can change dates once we have them

**Review by jmthibault79 — APPROVED** [review]
Works as expected: when the flag is true and my user hasn't completed training, I see the blocked message and I can't click.
When the flag is false or I have been bypassed (didn't test **completed**) then I see the module as normal.

File: ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx [hunk 3]
```diff
@@ -132,29 +159,32 @@ describe(ComplianceTrainingModuleCardTitle.name, () => {
...
         complianceTrainingRenewalLookback: 30,
       }
     );
 
-    await expectHelpTextToNotExist();
+    expectMaintenanceTextToExist();
+    expectHelpTextToNotExist();
   });
 
-  it('does not show help text if absorb is not used', async () => {
```
Comment (evrii) [inline ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx:157]: Duplicate test "does not show help text if module is bypassed" and "does not show help text if module is bypassed"

File: ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx [hunk 4]
```diff
@@ -25,25 +25,37 @@ import { serverConfigStore } from 'app/utils/stores';
...
 const expectHelpTextToNotExist = async () => {
-  try {
-    await findHelpText();
-    // The above throws an error if the element is not found
-  } catch (e) {
-    // Expected behavior, do nothing
-    return;
-  }
-  expect(true).toBeFalsy();
+  expect(queryHelpText()).not.toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (evrii) [reply ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx:55]: Great point!

I took a different approach. When checking for the absence of helper or maintenance text, I ensure that if the other text (maintenance or absence) is present first. If neither are supposed to be present, I check for the title. This ensures that the component has loaded first before checking the absence of text. What do you think about this approach?

[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/access/compliance-training-module-card-title.spec.tsx:55]: yes that should work.  thanks!
